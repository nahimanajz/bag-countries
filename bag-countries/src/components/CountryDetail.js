export function CountryDetail({children}){
    
    return(<div className="grid-2 margin-top-120">
    <div>            
         <img src="https://upload.wikimedia.org/wikipedia/en/thumb/b/ba/Flag_of_Germany.svg/1200px-Flag_of_Germany.svg.png" alt="Country flag"></img>
    </div>
    <div className="flex-col">
      <div className="title-dark">Belgium</div>
       <div className="grid-2">
         <ul>
           <li><b>Native Name:  </b> Belgie</li>
             <li><b>Population:  </b> 111.897.123</li>
             <li><b>Region:  </b> Europe</li>
             <li><b>Sub Region: </b>Western Europe</li>
             <li><b>Capital: </b>Brussels</li>
         </ul>
         <ul>
           <li><b>Top Level Domain:</b> .be</li>
             <li><b>Currencies:</b> 111.897.123</li>
             <li><b>Languages</b> Ducth, french, German</li> {/** TODO: map list of languages */}
         </ul>
       </div>
       <div className="container">
        <span><b>Border countries:</b></span>
         <span className="neighbor">France</span> 
         <span className="neighbor">Germany</span> <span className="neighbor">Netherlands</span>
      </div>
    </div>
    
  </div>

    )
}