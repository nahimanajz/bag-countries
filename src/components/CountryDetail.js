import { useEffect, useState } from "react";

export function CountryDetail({children, country, countries:allCountries}){
    const {flag, name, nativeName,population, region, subregion, capital, currencies,
        languages, topLevelDomain, borders
       } = country;
       const [neighbors, setNeighbors]= useState([]);
       useEffect(() => {  
           const co= []           
        Object.values(allCountries).map(({name, cioc, cca})=>                  
        borders && borders.find(border =>border === cioc?   
           co.push(name)             
        :''
        )
    )   
        setNeighbors(co)
      }, [neighbors]);
       
return( 
    <div className="flex-col">
        <div className ="mobile-menu">
             {children} 
        </div>
        
        <div className="info" key={name}>
            <div className="grid-2 margin-top-120">
                <div className="w-100">
                    <img className="flag"                 
                    src={flag}
                    alt="flag" />  
                </div>            
                       
               
                <div className="flex-col">
                <div className="title-dark">{name}</div>
                <div className="grid-2">
                    <ul>
                        <li><b>Native Name:  </b> {nativeName}</li>
                        <li><b>Population:  </b> {population.toLocaleString()}</li>
                        <li><b>Region:  </b> {region}</li>
                        <li><b>Sub Region: </b>{subregion}</li>
                        <li><b>Capital: </b>{capital}</li>
                    </ul>
                    <ul>
                         <li><b>Top Level Domain:</b> {topLevelDomain}</li>
                        <li><b>Currencies:</b> {currencies.map(currency => currency.name)}</li>
                        <li><b>Languages: </b> {languages && Object.values(languages).map(({name})=> <label key={name}> ,{name}  </label>)}</li> 
                </ul>
                <div className="container">
                <span><b>Border countries:</b></span>
                {
                  neighbors.map(neighbor=> <span className="neighbor">{neighbor}</span>) 
               }       

                </div>
                </div>
                
            </div>
        </div>
    </div>
    
</div>
    )
}