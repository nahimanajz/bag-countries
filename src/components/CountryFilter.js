
import { SearchIcon } from "../assets/SearchIcon";
export function CountryFilter(props){

    return(
        <div className="flex-wrap">
                    <div className="grid-col-2">
                        <span style={searchIcon}> <SearchIcon /> </span>                      
                      <input text="name" type="text"placeholder="Search for A Country" onChange={props.addCountryToList}/>
                    </div>
                    <div>
                        <select onChange={props.addCountryByRegion}>
                            <option>Filter by region</option>
                            <option>Africa</option>
                            <option value="Americas">America</option>
                            <option>Europe</option>
                            <option>Asia</option>
                            <option>Oceania</option>
                        </select>
                        
                    </div>
                </div>
    )
}

const searchIcon ={alignSelf: 'center', width: '24px'}