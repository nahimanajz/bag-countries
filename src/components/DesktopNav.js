import React, {useCallback} from 'react'
import {useNavigate } from "react-router-dom";

const { ArrowLeft } = require("../assets/ArrowLeft");
const { Bell } = require("../assets/Bell");
const { ToggleLeft } = require("../assets/ToggleLeft");
const { ToggleRight } = require("../assets/ToggleRight");


export function DesktopNav({toggleMode, darkMode, title,changeScreenTitle}){
    const navigate = useNavigate()
    const goBack = useCallback(
      () => {   
          navigate('/my-list')
          changeScreenTitle('My List')
      },[changeScreenTitle, navigate]
    ) 
    return(
        <div className="flex-wrap bottom-margin">
        <div className="flex desktop-back">              
            {title !== 'My List'&& (
              <span onClick={goBack}>
                  <ArrowLeft />
              </span>
            )}
        <span className="toggle">{title}</span>             
        </div>
        <div className="flex-end desktop">
          <div className="flex">
               <span> {darkMode?'Light Mode':'Dark Mode'}</span>                  
                <button onClick={toggleMode} className="toggle"> {darkMode ?<ToggleRight />:<ToggleLeft /> }</button>
          <div className="circle"><Bell color={'#fff'} /> </div>
          <div><b className="text-light">hey</b>, Jane</div>
          <div>
              <img className="icon"  src="https://data.whicdn.com/images/334652102/original.jpg" alt="user"></img>
          </div>

        </div>
        </div>            
      </div>
    )
}