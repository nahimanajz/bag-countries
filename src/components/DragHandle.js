import { DragIconWrapper } from "../util/DragIconWrapper";
import { Add } from "../assets/Add";
import React from "react";

export function DragHandle(props) {
  return (
    <DragIconWrapper {...props}>
      <Add />
    </DragIconWrapper>
  );
}
