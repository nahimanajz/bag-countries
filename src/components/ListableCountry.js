import { ToastContainer } from "react-toastify";
import { Add } from "../assets/Add";
import { Bin } from "../assets/Bin";
import { Check } from "../assets/Check";
export function ListableCountry({ countries, goToDetail, ...props }) {
  return (
    <>
      <ToastContainer />
      <div className="grid-col-3 m-top-32">
        {countries &&
          countries.map((country) => {
            const { name, capital, currencies, flag, population } = country;
            return (
              <div className="grid-row-4" key={name}>
                <div
                  className="flag"
                  onClick={() => goToDetail(country)}
                  style={{ backgroundImage: `url(${flag})` }}
                  alt="flag"
                />

                <ul className="subtitle2">
                  <strong>{name}</strong>
                </ul>

                <ul type="none">
                  <li>Population:{population}</li>
                  <li>Capital: {capital} </li>
                  <li>Currency: {currencies && currencies[0].code}</li>
                </ul>

                <div style={{ display: "flex", justifyContent: "flex-end" }}>
                  {props.add ? (
                    <span style={iconStyle} onClick={() => props.add(name)}>
                      <Add />
                    </span>
                  ) : (
                    <>
                      <span
                        style={iconStyle}
                        onClick={() => props.deleteCountry(name)}
                      >
                        <Bin />
                      </span>
                      <span
                        style={iconStyle}
                        onClick={() => props.updateCountry(name)}
                      >
                        <Check checked={props.checked && props.checked} />
                      </span>
                    </>
                  )}
                </div>
              </div>
            );
          })}
      </div>
    </>
  );
}
const iconStyle = { height: "64px", width: "64px" };
