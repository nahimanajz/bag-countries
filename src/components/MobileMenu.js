import { ArrowLeft } from "../assets/ArrowLeft";

const { Moon } = require("../assets/Moon");
const { Sun } = require("../assets/Sun");

export function MobileMenu({toggleMode, darkMode}){
    return(
        <div className="mobile">
            
            <div className="model-header mobile"> 
                <span className="title">Where in the world?</span>
                
                <span className="flex-wrap">
                    <span onClick={toggleMode} className="margin-right-32"> {!darkMode ?<Sun/>:<Moon /> }</span>
                <label>{!darkMode? 'Light Mode': 'Dark Mode'}</label> 
                </span>
            </div>
             
            <div className="start">              
                <ArrowLeft />
                <span>Back</span> 
            </div>
        </div>
    )
}