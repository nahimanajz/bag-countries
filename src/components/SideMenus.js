import { Link } from "react-router-dom";
export default function SideMenus({darkMode,style}) {
  const menuColor ={color:darkMode&&'tomato'}

    return (
      <div className="sidebar" style={style}>
          <ul className="side-menu padding-32">      
              <li>
                  <Link to="/checkList" className="menu-item" style={menuColor}>CheckList</Link>
              </li>
              <li>
                  <Link to="/my-list" className="menu-item" style={menuColor}>My List</Link>
              </li>
              <li>
                
                <Link to="/visited" className="menu-item" style={menuColor}>Visited</Link>
                </li>
              <li>
              <Link to="/toVisit" className="menu-item" style={menuColor}>To Visit</Link>
              </li>
           </ul>
      </div>
      
    );
}