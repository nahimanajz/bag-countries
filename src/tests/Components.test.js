import React from 'react';
import { shallow } from 'enzyme';
import { CountryDetail} from '../components/CountryDetail'
import { CountryFilter} from '../components/CountryFilter'
import { DesktopNav } from '../components/DesktopNav';
import { ListableCountry } from '../components/ListableCountry';
import { MobileMenu } from '../components/MobileMenu';
import SideMenus from '../components/SideMenus';

describe("Components different components", () => {
    it("should render country detail  component without error", () => {
      const wrapper = shallow(<CountryDetail />);
      expect(wrapper.getElements()).toMatchSnapshot();
    });
    it("should render country filter component without error", () => {
        const wrapper = shallow(<CountryFilter />);
        expect(wrapper.getElements()).toMatchSnapshot();
      });
      it("should render  desktop nav  without error", () => {
        const wrapper = shallow(<DesktopNav />);
        expect(wrapper.getElements()).toMatchSnapshot();
      });
      it("should render  Listable countrywithout error", () => {
        const wrapper = shallow(<ListableCountry />);
        expect(wrapper.getElements()).toMatchSnapshot();
      });
      it("should render  Mobile Menu without error", () => {
        const wrapper = shallow(<MobileMenu />);
        expect(wrapper.getElements()).toMatchSnapshot();
      });
      it("should render  SideMenu without error", () => {
        const wrapper = shallow(<SideMenus />);
        expect(wrapper.getElements()).toMatchSnapshot();
      });
})