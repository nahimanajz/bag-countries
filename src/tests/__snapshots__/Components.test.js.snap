// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components different components should render  Listable countrywithout error 1`] = `
Array [
  <React.Fragment>
    <ToastContainer
      autoClose={5000}
      closeButton={[Function]}
      closeOnClick={true}
      draggable={true}
      draggableDirection="x"
      draggablePercent={80}
      hideProgressBar={false}
      newestOnTop={false}
      pauseOnFocusLoss={true}
      pauseOnHover={true}
      position="top-right"
      role="alert"
      rtl={false}
      theme="light"
      transition={[Function]}
    />
    <div
      className="grid-col-3 m-top-32"
    />
  </React.Fragment>,
]
`;

exports[`Components different components should render  Mobile Menu without error 1`] = `
Array [
  <div
    className="mobile"
  >
    <div
      className="model-header mobile"
    >
      <span>
        Where in the world?
      </span>
      <span
        className="flex-wrap"
      >
        <span
          className="margin-right-32"
        >
           
          <Sun />
        </span>
        <label>
          Light Mode
        </label>
      </span>
    </div>
    <div
      className="start"
    >
      <ArrowLeft />
      <span>
        Back
      </span>
    </div>
  </div>,
]
`;

exports[`Components different components should render  SideMenu without error 1`] = `
Array [
  <div
    className="sidebar desktop"
  >
    <ul
      className="side-menu padding-32"
    >
      <li>
        <Link
          className="menu-item"
          style={
            Object {
              "color": undefined,
            }
          }
          to="/checkList"
        >
          CheckList
        </Link>
      </li>
      <li>
        <Link
          className="menu-item"
          style={
            Object {
              "color": undefined,
            }
          }
          to="/my-list"
        >
          My List
        </Link>
      </li>
      <li>
        <Link
          className="menu-item"
          style={
            Object {
              "color": undefined,
            }
          }
          to="/visited"
        >
          Visited
        </Link>
      </li>
      <li>
        <Link
          className="menu-item"
          style={
            Object {
              "color": undefined,
            }
          }
          to="/toVisit"
        >
          To Visit
        </Link>
      </li>
    </ul>
  </div>,
]
`;

exports[`Components different components should render country filter component without error 1`] = `
Array [
  <div
    className="flex-wrap"
  >
    <div
      className="grid-col-2"
    >
      <span
        style={
          Object {
            "alignSelf": "center",
            "width": "24px",
          }
        }
      >
         
        <SearchIcon />
         
      </span>
      <input
        placeholder="Search for A Country"
        text="name"
        type="text"
      />
    </div>
    <div>
      <select>
        <option>
          Filter by region
        </option>
        <option>
          Africa
        </option>
        <option
          value="Americas"
        >
          America
        </option>
        <option>
          Europe
        </option>
        <option>
          Asia
        </option>
        <option>
          Oceania
        </option>
      </select>
    </div>
  </div>,
]
`;
